primitives/beefy/src/witness.rs

```diff
@@ -81,15 +84,30 @@ mod tests {
 	use super::*;
 	use codec::Decode;
 
-	use crate::{crypto, known_payloads, Payload, KEY_TYPE};
+	use crate::{
+		bls_crypto::Signature as BLSSignature, ecdsa_crypto, known_payloads, Payload, KEY_TYPE,
+	};
+	use bls_like::{
+		pop::SignatureAggregatorAssumingPoP, EngineBLS, SerializableToBytes, Signed, BLS377,
+	};
```

Comment:
I see `pub type BLS377 = UsualBLS<ark_bls12_377::Bls12_377, ark_bls12_377::Config>;` and `UsualBLS` uses 48 byte keys and hashes-to-G2, but..

We do not want to hash-to-G2.  We want to both hash-to-G1 and also run apk proofs on G1.

This requires that public keys be on both G1 and G2 and contain a DLEQ proof between them.  This is the point of https://eprint.iacr.org/2022/1611